File: pet_hospital_frontend/src/api/registration.ts
import api from './index'
import type { PageResult } from './customer'

export interface Registration {
    id?: number
    customerId: number
    petId: number
    doctorId?: number
    status?: 'PENDING' | 'WAITING' | 'IN_PROGRESS' | 'COMPLETED' | 'CANCELLED'
    type?: 'APPOINTMENT' | 'ONSITE'
    appointmentTime?: string
    queueNumber?: number
    createdAt?: string
    updatedAt?: string
    // 关联字段
    customer?: {
        name?: string
        phone?: string
    }
    pet?: {
        name?: string
        species?: string
    }
    doctor?: {
        name?: string
    }
}

export const registrationApi = {
    list: (params: { status?: string; date?: string; page?: number; size?: number }) =>
        api.get<any, { code: number; data: PageResult<Registration> }>('/registrations', { params }),

    getById: (id: number) =>
        api.get<any, { code: number; data: Registration }>(`/registrations/${id}`),

    create: (data: Registration) =>
        api.post<any, { code: number; data: Registration }>('/registrations', data),

    update: (id: number, data: Partial<Registration>) =>
        api.put<any, { code: number; data: boolean }>(`/registrations/${id}`, data),

    cancel: (id: number) =>
        api.delete<any, { code: number; data: boolean }>(`/registrations/${id}`),

    getQueueBoard: () =>
        api.get<any, { code: number; data: Registration[] }>('/registrations/queue'),

    callNext: (id: number) =>
        api.put<any, { code: number; data: boolean }>(`/registrations/${id}/call`),

    complete: (id: number) =>
        api.put<any, { code: number; data: boolean }>(`/registrations/${id}/complete`)
}

File: pet_hospital_backend/src/main/java/com/pethospital/entity/Registration.java
package com.pethospital.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("registrations")
public class Registration {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long customerId;
    private Long petId;
    private Long doctorId;
    
    /**
     * 状态: PENDING(待确认), WAITING(候诊中), IN_PROGRESS(就诊中), COMPLETED(已完成), CANCELLED(已取消)
     */
    private String status;
    
    /**
     * 类型: APPOINTMENT(预约), ONSITE(现场挂号)
     */
    private String type;
    
    @com.fasterxml.jackson.annotation.JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime appointmentTime;
    
    /**
     * 排队号 (当天有效)
     */
    private Integer queueNumber;
    
    @TableField(fill = FieldFill.INSERT)
    @com.fasterxml.jackson.annotation.JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    @com.fasterxml.jackson.annotation.JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updatedAt;
    
    // 非数据库字段，用于关联查询
    @TableField(exist = false)
    private Customer customer;
    
    @TableField(exist = false)
    private Pet pet;
    
    @TableField(exist = false)
    private Employee doctor;
}
